apps/client/src/app/modules/freecompany-workshops/+state/freecompany-workshop.facade.ts

```diff
@@ -0,0 +1,325 @@
+          submarine.dest5
+        ].filter((dest) => dest > 0)
+      };
+    })),
+    shareReplay(1)
+  );
+
+  constructor(private readonly lazyData: LazyDataService, private readonly ipc: IpcService,
+              private readonly store: Store<fromFreecompanyWorkshop.State>) {
+  }
+
+  public isSubmarineItemInfo(itemInfo: ItemInfo | UpdateInventorySlot): boolean {
+    return itemInfo.containerId === 25004 && itemInfo.slot <= 18;
+  }
+
+  public isAirshipItemInfo(itemInfo: ItemInfo | UpdateInventorySlot): boolean {
+    return itemInfo.containerId === 25003 && itemInfo.slot >= 30 && itemInfo.slot <= 48;
+  }
+
+  public load(): void {
+    this.store.dispatch(FreecompanyWorkshopActions.readFromFile());
+  }
+
+  public setCurrentFreecompanyId(id: string) {
+    this.store.dispatch(FreecompanyWorkshopActions.setFreecompanyId({ id }));
+  }
+
+  public importFromPcap(): void {
+    this.store.dispatch(FreecompanyWorkshopActions.importFromPcap());
+  }
+
+  public getVesselPartCondition(itemInfo: ItemInfo | UpdateInventorySlot): { type: VesselType, vesselSlot: number, partSlot: number, condition: number } {
+    const partCondition: any = {
+      condition: itemInfo.condition
+    };
+
+    const airshipSlots = [
+      [30, 31, 32, 33], // hull
+      [35, 36, 37, 38], // rigging
+      [40, 41, 42, 43], // forecastle
+      [45, 46, 47, 48] // aftcastle
+    ];
+    const submarineSlots = [
+      [0, 1, 2, 3], // hull
+      [5, 6, 7, 8], // stern
+      [10, 11, 12, 13], // bow
+      [15, 16, 17, 18] // bridge
+    ];
+
+    if (itemInfo.containerId === 25003) {
+      partCondition.type = VesselType.AIRSHIP;
+      for (let i = 0; i < 4; i++) {
+        if (airshipSlots[i].includes(itemInfo.slot)) {
+          partCondition.vesselSlot = i;
+          partCondition.partSlot = airshipSlots[i].indexOf(itemInfo.slot);
+          break;
+        }
+      }
+    } else if (itemInfo.containerId === 25004) {
+      partCondition.type = VesselType.SUBMARINE;
+      for (let i = 0; i < 4; i++) {
+        if (submarineSlots[i].includes(itemInfo.slot)) {
+          partCondition.vesselSlot = i;
+          partCondition.partSlot = submarineSlots[i].indexOf(itemInfo.slot);
+          break;
+        }
+      }
+    }
+
+    return partCondition.type !== undefined ? partCondition : null;
+  }
+
+  public updateVesselPartCondition(packet: ItemInfo | UpdateInventorySlot): void {
+    const partUpdate = this.getVesselPartCondition(packet);
+    this.store.dispatch(FreecompanyWorkshopActions.updateVesselPart({ vesselPartUpdate: partUpdate }));
+  }
+
+  public updateVesselTimers(data): void {
+    this.store.dispatch(FreecompanyWorkshopActions.updateVesselTimers({
+      vesselTimersUpdate: {
+        type: data.type,
+        timers: data.timers
+      }
+    }));
+  }
+
+  public getRemainingTime(unixTimestamp) {
+    return unixTimestamp - Math.floor(Date.now() / 1000);
+  }
+
+  public getSubmarineBuild(submarine: Submarine) {
```

Comment:
A proper type interface for this would be nice, as it's used in other places, and lazy data isn't typed at all for now.